tests/gate_data.py

```diff
@@ -0,0 +1,177 @@
+"""Convenience gate representations for testing"""
+import numpy as np
+
+# ========================================================
+#  fixed gates
+# ========================================================
+
+I = np.eye(2)
+# Pauli matrices
+X = np.array([[0, 1], [1, 0]])  #: Pauli-X matrix
+Y = np.array([[0, -1j], [1j, 0]])  #: Pauli-Y matrix
+Z = np.array([[1, 0], [0, -1]])  #: Pauli-Z matrix
+
+H = np.array([[1, 1], [1, -1]]) / np.sqrt(2)  #: Hadamard gate
+# Two qubit gates
+CNOT = np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 1], [0, 0, 1, 0]])  #: CNOT gate
+SWAP = np.array([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]])  #: SWAP gate
+CZ = np.array([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, -1]])  #: CZ gate
+S = np.array([[1, 0], [0, 1j]])  #: Phase Gate
+T = np.array([[1, 0], [0, np.exp(1j * np.pi / 4)]])  #: T Gate
+# Three qubit gates
+CSWAP = np.array(
+    [
+        [1, 0, 0, 0, 0, 0, 0, 0],
+        [0, 1, 0, 0, 0, 0, 0, 0],
+        [0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 1, 0, 0, 0, 0],
+        [0, 0, 0, 0, 1, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 1, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 1],
+    ]
+)  #: CSWAP gate
+
+Toffoli = np.diag([1 for i in range(8)])
+Toffoli[6:8, 6:8] = np.array([[0, 1], [1, 0]])
+
+# ========================================================
+#  parametrized gates
+# ========================================================
+
+
+def Rphi(phi):
```

Comment:
In principle, one could also make those methods simpler by putting
```python
    def Rphi(phi):
        return qml.PhaseShift(phi).matrix
```
But I guess the better solution would just the regex replace `Rphi(phi)` with `qml.PhaseShift(phi).matrix`.

Reply:
Yep, I thought about this. The reason why I didn't: I thought it would be best to treat `gate_data.py` as containing reference data for the tests, that is independent of the PennyLane source code (i.e., a 'source of truth').

But I am not 100% there, and am very tempted by the fact that your proposal could remove more redundant code! What are your thoughts?

Reply:
Yeah, I think that is a very good argument, we should keep that as an independent source of matrices!